astroid/inference.py

```diff
@@ -193,7 +195,7 @@ def _infer_map(
             if any(not elem for elem in (key, safe_value)):
                 raise InferenceError(node=node, context=context)
             # safe_value is SuccessfulInferenceResult as bool(Uninferable) == False
-            values = _update_with_replacement(values, {key: safe_value})  # type: ignore[dict-item]
+            values = _update_with_replacement(values, {key: safe_value})
```

Comment:
Have you tested this with ``mypy --strict``? I added these ignores based on strict mode so they might not be unused in that case.

Reply:
Yes, with version `mypy 0.991 (compiled: yes)` and `Python 3.8.10`. It warned that only some of the ignores are unused. I guess some have the nodes have the function types declared and some don't?